Source/JavaScriptCore/JavaScriptCore_Private.modulemap

```diff
@@ -0,0 +1,93 @@
+framework module JavaScriptCore_Private [system] {
+    // These ad-hoc explicit modules are the exhaustive set of headers that are both private and compatible
+    // with both C and C++.
+
+    explicit module APIIntegrityPrivate {
+        header "APIIntegrityPrivate.h"
+        export *
+    }
+
+    explicit module JSBasePrivate {
+        header "JSBasePrivate.h"
+        export *
+    }
+
+    explicit module JSContextRefPrivate {
+        header "JSContextRefPrivate.h"
+        export *
+    }
+
+    explicit module JSHeapFinalizerPrivate {
+        header "JSHeapFinalizerPrivate.h"
+        export *
+    }
+
+    explicit module JSLockRefPrivate {
+        header "JSLockRefPrivate.h"
+        export *
+    }
+
+    explicit module JSMarkingConstraintPrivate {
+        header "JSMarkingConstraintPrivate.h"
+        export *
+    }
+
+    explicit module JSObjectRefPrivate {
+        header "JSObjectRefPrivate.h"
+        export *
+    }
+
+    explicit module JSRemoteInspector {
+        header "JSRemoteInspector.h"
+        export *
+    }
+
+    explicit module JSScriptRefPrivate {
+        header "JSScriptRefPrivate.h"
+        export *
+    }
+
+    explicit module JSStringRefPrivate {
+        header "JSStringRefPrivate.h"
+        export *
+    }
+
+    explicit module JSWeakPrivate {
+        header "JSWeakPrivate.h"
+        export *
+    }
+
+    explicit module PASReportCrashPrivate {
+        header "PASReportCrashPrivate.h"
+        export *
+    }
+
+    explicit module Cxx {
```

Comment:
FWIW I've started to move away from using the keyword `explicit`. I'm still not sure of the pros and cons, but I've found no advantages to using the keyword, and presumably it might require somebody in future to do something like `import JavaScriptCore.Cxx` which is probably undesirable.